### PR TITLE
Feature/43332 move pupil la codes to db

### PR DESCRIPTION
**Pull request by jon-shipley**
Move LA Codes to DB
Mod pupil UPN validator to use DB to obtain LA codes

### CODE REVIEW

File: admin/lib/validator/pupil-validator.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ module.exports.validate = async (pupilData, schoolId, isMultiplePupilsSubmission
   if (!isEmpty(upn) && expected !== upn[0]) {
     upnErrorArr.push(addPupilErrorMessages.upnInvalidCheckDigit)
   }
-  if (!isEmpty(upn) && !upnService.hasValidLaCode(upn)) {
+  const isValidLaCode = await upnService.hasValidLaCode(upn)
+  if (!isEmpty(upn) && !isValidLaCode) {
```
Comment (GuyHarwood) [inline admin/lib/validator/pupil-validator.js:137]: wouldn't it be less expensive to test for empty first?

File: db/migrations/core-data/20210129171943.do.populate-la-code-table.sql [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (926, N'Norfolk');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (928, N'Northamptonshire');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (929, N'Northumberland');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (931, N'Oxfordshire');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (933, N'Somerset');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (935, N'Suffolk');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (936, N'Surrey');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (937, N'Warwickshire');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (938, N'West Sussex');
+INSERT INTO mtc_admin.laCodeLookup (laCode, laName) VALUES (999, N'Test');
```
Comment (GuyHarwood) [inline db/migrations/core-data/20210129171943.do.populate-la-code-table.sql:185]: blank new line required for end of file

**Review by GuyHarwood — APPROVED** [review]
LGTM, just a couple of minor points

File: admin/lib/validator/pupil-validator.js (hunk 1 above)
Comment (jon-shipley) [reply admin/lib/validator/pupil-validator.js:137]: I don't think so.  All the tests are applied before returning the results to the user.

Comment (GuyHarwood) [reply admin/lib/validator/pupil-validator.js:137]: yep, just opened the file and can see that now. thanks

**Pull request closed without merging**